presto-raptorx/src/main/java/com/facebook/presto/raptorx/metadata/MasterWriterDao.java

```diff
@@ -0,0 +1,323 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.raptorx.metadata;
+
+import org.jdbi.v3.sqlobject.customizer.Bind;
+import org.jdbi.v3.sqlobject.statement.SqlQuery;
+import org.jdbi.v3.sqlobject.statement.SqlUpdate;
+import org.jdbi.v3.sqlobject.statement.UseRowMapper;
+
+import java.util.List;
+
+public interface MasterWriterDao
+{
+    // transaction
+
+    @SqlQuery("SELECT commit_id FROM current_commit FOR UPDATE")
+    Long getLockedCurrentCommitId();
+
+    @SqlUpdate("INSERT INTO active_commit (singleton, commit_id, start_time, rolling_back)\n" +
+            "VALUES ('X', :commitId, :startTime, FALSE)")
+    void insertActiveCommit(
+            @Bind long commitId,
+            @Bind long startTime);
+
+    @SqlQuery("SELECT * FROM active_commit")
+    @UseRowMapper(ActiveCommit.Mapper.class)
+    ActiveCommit getActiveCommit();
+
+    @SqlUpdate("DELETE FROM active_commit")
+    int deleteActiveCommit();
+
+    @SqlUpdate("UPDATE active_commit SET rolling_back = TRUE")
+    void abortActiveCommit();
+
+    @SqlUpdate("UPDATE active_commit SET rollback_info = :rollbackInfo")
+    void updateRollbackInfo(
+            @Bind byte[] rollbackInfo);
+
+    @SqlUpdate("INSERT INTO commits (commit_id, commit_time)\n" +
```

Comment:
Do we need this "commits" table? I see we insert and delete it, but it's not read, right?

Reply:
It can be helpful for debugging to know when a commit happened. Also, it allows features such as rolling back the database to a specific point in time.